server/src/internalClusterTest/java/org/elasticsearch/indices/DataStreamIT.java

```diff
@@ -592,6 +605,58 @@ public void testTimestampFieldCustomAttributes() throws Exception {
         client().admin().indices().deleteDataStream(deleteDataStreamRequest).actionGet();
     }
 
+    public void testUpdateMappingViaDataStream() throws Exception {
+        putComposableIndexTemplate("id1", "@timestamp", List.of("logs-*"));
+        CreateDataStreamAction.Request createDataStreamRequest = new CreateDataStreamAction.Request("logs-foobar");
+        client().admin().indices().createDataStream(createDataStreamRequest).actionGet();
+
+        String backingIndex1 = DataStream.getDefaultBackingIndexName("logs-foobar", 1);
+        String backingIndex2 = DataStream.getDefaultBackingIndexName("logs-foobar", 2);
+
+        RolloverResponse rolloverResponse = client().admin().indices().rolloverIndex(new RolloverRequest("logs-foobar", null)).get();
+        assertThat(rolloverResponse.getNewIndex(), equalTo(backingIndex2));
+        assertTrue(rolloverResponse.isRolledOver());
+
+        Map<?, ?> expectedMapping = Map.of("properties", Map.of("@timestamp", Map.of("type", "date")));
+        GetMappingsResponse getMappingsResponse = getMapping("logs-foobar").get();
+        assertThat(getMappingsResponse.getMappings().size(), equalTo(2));
+        assertThat(getMappingsResponse.getMappings().get(backingIndex1).getSourceAsMap(), equalTo(expectedMapping));
+        assertThat(getMappingsResponse.getMappings().get(backingIndex2).getSourceAsMap(), equalTo(expectedMapping));
+
+        expectedMapping = Map.of("properties", Map.of("@timestamp", Map.of("type", "date"), "my_field", Map.of("type", "keyword")));
+        putMapping("{\"properties\":{\"my_field\":{\"type\":\"keyword\"}}}", "logs-foobar").get();
+        // Only the mapping of th latest backing index should be updated:
```

Comment:
Is this comment correct? It looks like it applies the mapping to all backing indices.

Reply:
👍 let me fix that